torchserve/datadog_checks/torchserve/model_discovery.py

```diff
@@ -0,0 +1,63 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.checks.kubelet_base.base import urljoin
+from datadog_checks.base.utils.discovery import Discovery
+
+
+class ModelDiscovery(Discovery):
+    def __init__(self, check, limit=None, include=None, exclude=None, interval=None):
+        super().__init__(
+            self.get_models,
+            limit=limit,
+            include={pattern: None for pattern in include} if include else None,
+            exclude=exclude,
+            interval=interval,
+            key=lambda n: n.get("modelName"),
+        )
+        self.check = check
+        self.api_status = AgentCheck.UNKNOWN
+        self.all_models = None
+        self.refreshed = False
+
+    def get_items(self):
+        self.refreshed = False
+        return super().get_items()
+
+    def get_models(self):
+        self.refreshed = True
+        page_number = 0
+        models = []
+
+        try:
+            url = urljoin(self.check.instance["management_api_url"], "models")
+            self.check.log.debug("Querying page [%d]", page_number)
+            response = self.check.http.get(url, params={"limit": 100})
```

Comment:
Just curious, is the value of this `limit` supposed to be the same as the `ModelDiscovery`'s `limit`?

Reply:
Torchserve will return up to 100 models at a time. We need to iterate over all of them to get them and after that we use our own `limit` with the include/exclude options so no, it's not the same